ShellPkg/Library/UefiShellDriver1CommandsLib/DrvDiag.c

```diff
@@ -199,17 +199,24 @@ DoDiagnostics (
                               );
               if (!EFI_ERROR (Status) && (DriverDiagnostics2 != NULL)) {
                 Language = GetBestLanguageForDriver (DriverDiagnostics2->SupportedLanguages, Lang, FALSE);
-                Found    = TRUE;
-                Status   = DriverDiagnostics2->RunDiagnostics (
-                                                 DriverDiagnostics2,
-                                                 ControllerHandleList[ControllerHandleListLoop],
-                                                 ChildHandleList == NULL ? NULL : ChildHandleList[ChildHandleListLoop],
-                                                 (EFI_DRIVER_DIAGNOSTIC_TYPE)Mode,
-                                                 Language,
-                                                 &ErrorType,
-                                                 &OutBufferSize,
-                                                 &OutBuffer
-                                                 );
+                // MU_CHANGE [START] - CodeQL change
+                if (Language == NULL) {
+                  Language = AllocateCopyPool (StrSize (L"en-US"), L"en-US");

```

Comment:
This would seem to be different behavior where we fall back to english (i.e. more than just a CodeQL fix). Do we want this new behavior or to fail? If we do, we should check that AllocateCopyPool succeeded.